src/plugins/uploader.js

```diff
@@ -0,0 +1,154 @@
+"use strict";
+
+const SocketIOFileUploadServer = require("socketio-file-upload/server");
+const Helper = require("../helper");
+const path = require("path");
+const fsextra = require("fs-extra");
+const fs = require("fs");
+const fileType = require("file-type");
+const readChunk = require("read-chunk");
+const crypto = require("crypto");
+const isUtf8 = require("is-utf8");
+
+const whitelist = [
+	"text/plain",
+
+	"image/png",
+	"image/jpeg",
+	"image/jpg",
+	"image/gif",
+	"image/bmp",
+	"image/webp",
+
+	"audio/midi",
+	"audio/x-midi",
+	"video/mp4",
+	"audio/mpeg",
+	"audio/mpeg3",
+	"audio/x-mpeg",
+	"audio/x-mpeg-3",
+	"audio/wave",
+	"audio/wav",
+	"audio/x-wav",
+	"audio/x-pn-wav",
+	"audio/webm",
+	"video/webm",
+	"audio/ogg",
+	"video/ogg ",
+	"application/ogg",
+];
+
+class Uploader {
+	constructor(client, socket) {
+		const siofu = new SocketIOFileUploadServer();
+
+		siofu.uploadValidator = function(event, callback) {
+			const folder = path.join(Helper.getFileUploadPath(), ".tmp");
+			fsextra.ensureDir(folder, (err) => {
+				if (err) {
+					callback(err);
```

Comment:
Any reason this folder creator is in `uploadValidator` and not simply in the constructor?

And looking at the library code, err shouldn't be passed through to correctly abort the error: 
https://github.com/sffc/socketio-file-upload/blob/c1946fe3e43bd0692100c350c93771ec01ac0823/server.js#L314-L316

Reply:
No, except it exists in the library for basically this purpose. I can do either way

Reply:
I think we will be fine just creating the folder in the constructor.

Instead of error callback, print a warning to the console.

Reply:
if you want a warning in console on error, do you mean I should leave the `ensureDir` in the uploadValidator? If we're creating the directory in the constructor, then no need to override uploadValidator.

Reply:
Nevermind, got it

Reply:
done